hack/generator/pkg/codegen/pipeline_improve_resource_pluralization.go

```diff
@@ -20,12 +20,21 @@ func improveResourcePluralization() PipelineStage {
 
 			result := make(astmodel.Types)
 			for _, typeDef := range types {
-				if _, ok := typeDef.Type().(*astmodel.ResourceType); ok {
+				if resourceType, ok := typeDef.Type().(*astmodel.ResourceType); ok {
 					newTypeName := typeDef.Name().Singular()
 					// check if there is already a resource with this name
 					if _, ok := types[newTypeName]; !ok {
 						// not found: rename the resource
-						result.Add(typeDef.WithName(newTypeName))
+						typeDef = typeDef.WithName(newTypeName)
+
+						// Need to update owner ref too if applicable
+						if resourceType.Owner() != nil {
```

Comment:
Do we also need to change the spec type's name (and eventually the status type's) to match? It looks a bit weird in the generated code to have the top-level resource type singular but the spec plural. 
Could be worth adding a golden test for pluralisation? (Actually I think I have the start of one on my machine but it got bogged down in properly representing resources in the json files, which it seems like you've solved.)

Reply:
> It looks a bit weird in the generated code to have the top-level resource type singular but the spec plural.
I agree that this is a bit weird - but this was the behavior of pluralization prior to this PR too. I think @theunrepentantgeek fought with this some and doing it at the resource level was the only way that didn't result in tons of issues. Maybe resource + spec is ok (since the spec name is derived from the resource name)?

Probably something for after this PR though.

I created: https://github.com/Azure/k8s-infra/issues/220 tracking that.